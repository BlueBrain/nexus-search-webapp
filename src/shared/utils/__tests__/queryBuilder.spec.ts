import makeDocQuery from '../queryBuilder';

describe('makeDocQuery()', () => {
  it(' makes an elasticSearch agg object, and the proper with @types, terms, and field', () => {
    expect(makeDocQuery({})).toMatchSnapshot();
  });

  it(' makes an elasticSearch aggs object with a query if a textQuery argument is provided', () => {
    const textQuery = { q: 'mySearchString' };
    expect(makeDocQuery(textQuery)).toMatchSnapshot();
  });

  it(' makes an elasticSearch aggs object with a bool must match term if a type argument is provided', () => {
    const textAndTypeQuery = { q: 'mySearchString' };
    expect(makeDocQuery(textAndTypeQuery)).toMatchSnapshot();
  });

  it(' makes an elasticSearch aggs object even with filters provided!', () => {
    const complexQuery = {
      q: 'mySearchString',
      filter: { 'sex.labels': ['male'] },
    };
    expect(makeDocQuery(complexQuery)).toMatchSnapshot();
  });

  it(' makes an elasticSearch aggs object even with nested filters', () => {
    const nestedQuery = {
      q: 'mySearchString',
      filter: {
        'brainLocation.brainRegion.labels': [
          'Posterior complex of the thalamus',
        ],
      },
    };
    expect(makeDocQuery(nestedQuery)).toMatchSnapshot();
  });
});
