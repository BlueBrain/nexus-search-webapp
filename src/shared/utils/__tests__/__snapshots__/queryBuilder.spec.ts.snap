// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeDocQuery()  makes an elasticSearch agg object, and the proper with @types, terms, and field 1`] = `
Object {
  "query": Object {
    "bool": Object {
      "must": Array [
        Object {
          "term": Object {
            "_deprecated": false,
          },
        },
      ],
    },
  },
}
`;

exports[`makeDocQuery()  makes an elasticSearch aggs object even with filters provided! 1`] = `
Object {
  "query": Object {
    "bool": Object {
      "must": Array [
        Object {
          "term": Object {
            "_deprecated": false,
          },
        },
        Object {
          "match": Object {
            "_all_fields": "mySearchString",
          },
        },
        Object {
          "bool": Object {
            "must": Array [
              Object {
                "nested": Object {
                  "path": "sex",
                  "query": Object {
                    "bool": Object {
                      "should": Array [
                        Object {
                          "term": Object {
                            "sex.labels": "male",
                          },
                        },
                      ],
                    },
                  },
                },
              },
            ],
          },
        },
      ],
    },
  },
}
`;

exports[`makeDocQuery()  makes an elasticSearch aggs object even with nested filters 1`] = `
Object {
  "query": Object {
    "bool": Object {
      "must": Array [
        Object {
          "term": Object {
            "_deprecated": false,
          },
        },
        Object {
          "match": Object {
            "_all_fields": "mySearchString",
          },
        },
        Object {
          "bool": Object {
            "must": Array [
              Object {
                "nested": Object {
                  "path": "brainLocation",
                  "query": Object {
                    "nested": Object {
                      "path": "brainLocation.brainRegion",
                      "query": Object {
                        "bool": Object {
                          "should": Array [
                            Object {
                              "term": Object {
                                "brainLocation.brainRegion.labels": "Posterior complex of the thalamus",
                              },
                            },
                          ],
                        },
                      },
                    },
                  },
                },
              },
            ],
          },
        },
      ],
    },
  },
}
`;

exports[`makeDocQuery()  makes an elasticSearch aggs object with a bool must match term if a type argument is provided 1`] = `
Object {
  "query": Object {
    "bool": Object {
      "must": Array [
        Object {
          "term": Object {
            "_deprecated": false,
          },
        },
        Object {
          "match": Object {
            "_all_fields": "mySearchString",
          },
        },
      ],
    },
  },
}
`;

exports[`makeDocQuery()  makes an elasticSearch aggs object with a query if a textQuery argument is provided 1`] = `
Object {
  "query": Object {
    "bool": Object {
      "must": Array [
        Object {
          "term": Object {
            "_deprecated": false,
          },
        },
        Object {
          "match": Object {
            "_all_fields": "mySearchString",
          },
        },
      ],
    },
  },
}
`;
